src/Integration.Vsix/CFamily/ProcessRunner.cs

```diff
@@ -52,7 +53,7 @@ public ProcessRunner(ISonarLintSettings settings, ILogger logger)
         /// Runs the specified executable and returns a boolean indicating success or failure
```

Comment:
Most of this comment is no longer accurate. More specifically, the threading implications of how the process is executed are not at all obvious so it needs some documentation to help us maintain it in future (not necessarily in a doc comment for this method - I think it needs more detail than that).

Some of the important points to capture (you can probably think of more):
* calling Execute is a blocking call. The method will return after the timeout or when the handler has read to the end of the output stream.
* the stream reader callbacks are executed on the original calling thread (?)
* we're using two threads: one to read the data from the stream, one to handle the timeout.
* errors and timeouts are logged to the output window. The caller won't see them and has no way of checking the outcome.